sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -360,22 +372,24 @@ trait DivModLike extends BinaryArithmetic {
         boolean ${ev.isNull} = false;
         $javaType ${ev.value} = ${CodeGenerator.defaultValue(dataType)};
         if ($isZero) {
-          ${ev.isNull} = true;
+          $divByZero
         } else {
           ${eval1.code}
           ${ev.value} = $operation;
         }""")
     } else {
       ev.copy(code = code"""
-        ${eval2.code}
```

Comment:
we should run `eval2` first, as we have a chance to skip `eval1` if `eval2` is 0.

This is also why we didn't use `nullSafeEval` before, to control the evaluation order.

It would be great if you can help to write this down as code comments, so that other people won't make the same mistake in the future.

Reply:
done